tests/cpu-vm

```diff
@@ -86,35 +85,42 @@ done
 lxc config set v1 limits.cpu="${cpuCount}"
 [ "$(lxc exec v1 -- ls /sys/devices/system/cpu | grep -xEc 'cpu[[:digit:]]+')" -eq "${cpuCount}" ]
 
-# check that CPU affinity is automatically set if feature present
+echo "==> Check that CPU affinity is automatically set when limits.cpu.pin_strategy=auto"
 QEMU_PID=$(lxc info v1 | awk '/^PID:/ {print $2}')
-if journalctl --quiet --no-hostname --no-pager --boot=0 --unit=snap.lxd.daemon.service | grep "Scheduler: virtual-machine"; then
-  # Check that there are processes with pinning set
-  # It will be shown like this (for limits.cpu=2):
-  # pid 2894's current affinity list: 6
-  # pid 2895's current affinity list: 8
-  # pid 2897's current affinity list: 0-15
-  # pid 2898's current affinity list: 0-15
-  # pid 2899's current affinity list: 0-15
-  # 2894 and 2895 have affinity set, while others don't
-  PINNED_THREADS_NUM=$(taskset --cpu-list -a -p "${QEMU_PID}" | grep -cE ':\s+[0-9]+$')
-  [ "${PINNED_THREADS_NUM}" -ge "$(lxc config get v1 limits.cpu)" ]
-else
-  # check that there is no pinning set
-  ! taskset --cpu-list -a -p "${QEMU_PID}" | grep -E ':\s+[0-9]+$' || false
-  taskset --cpu-list -a -p "${QEMU_PID}" | grep "0-$((cpuCount-1))"
-fi
+# Check that there are processes with pinning set
+# It will be shown like this (for limits.cpu=2):
+# pid 2894's current affinity list: 6
+# pid 2895's current affinity list: 8
+# pid 2897's current affinity list: 0-15
+# pid 2898's current affinity list: 0-15
+# pid 2899's current affinity list: 0-15
+# 2894 and 2895 have affinity set, while others don't
+PINNED_THREADS_NUM=$(taskset --cpu-list -a -p "${QEMU_PID}" | grep -cE ':\s+[0-9]+$')
+[ "${PINNED_THREADS_NUM}" -ge "$(lxc config get v1 limits.cpu)" ]
 
 # Unset CPU limit
 lxc config unset v1 limits.cpu
 
 # Unsetting the limit should leave the VM with 1 CPU
 [ "$(lxc exec v1 -- ls /sys/devices/system/cpu | grep -xEc 'cpu[[:digit:]]+')" -eq "1" ]
 
-echo "==> Stopping and deleting ephemeral VM"
-# Stop VM and check its deleted.
+echo "==> Create ephemeral VM with CPU pinning disabled and boot"
+lxc launch "${IMAGE}" v2 --vm -c limits.cpu.pin_strategy=none -s "${poolName}" --ephemeral
```

Comment:
`limits.cpu.pin_strategy=none` being the intended default, I would probably not specify it and simply check that no pinning is applied with default settings. I'd also revert to the old `echo`.

Reply:
Also, I would try to minimize the number of running/existing instances if possible. As such, I'd keep using `v1` for as long as it make sense and stop + reconfig it if needed/possible. Otherwise, nuking it and creating a fresh `v1` is also possible.

Creating instances has a cost (resource+time) that's higher than reconfiguring an existing one.